pkg/route/registry/route/strategy.go

```diff
@@ -79,23 +83,26 @@ func (s routeStrategy) allocateHost(ctx apirequest.Context, route *routeapi.Rout
 		if !ok {
 			return field.ErrorList{field.InternalError(field.NewPath("spec", "host"), fmt.Errorf("unable to verify host field can be set"))}
 		}
-		res, err := s.sarClient.CreateSubjectAccessReview(
-			ctx,
-			authorizationapi.AddUserToSAR(
+		res, err := s.sarClient.Create(
+			authorizationutil.AddUserToSAR(
 				user,
 				&authorizationapi.SubjectAccessReview{
-					Action: authorizationapi.Action{
-						Verb:     "create",
-						Group:    routeapi.GroupName,
-						Resource: "routes/custom-host",
+					Spec: authorizationapi.SubjectAccessReviewSpec{
+						ResourceAttributes: &authorizationapi.ResourceAttributes{
+							Namespace:   apirequest.NamespaceValue(ctx),
+							Verb:        "create",
+							Group:       routeapi.GroupName,
+							Resource:    "routes",
+							Subresource: "custom-host",
```

Comment:
I know this is preexisting, but does it have some special meaning?

Reply:
> I know this is preexisting, but does it have some special meaning?

yeah, they used a subresource as an authorization check on a different field.